stock_picking_force_assign/models/stock_picking.py

```diff
@@ -48,6 +48,12 @@ def _force_assign_find_moves(self):
                 ('product_id', '=', move.product_id.id),
                 ('state', 'in', ('partially_available', 'assigned')),
             ], order='write_date asc')
+            if not candidates:
+                raise exceptions.UserError(
+                    _('There is no sufficient stock of product: %s in location: %s') % (
+                        move.product_id.display_name, move.location_id.display_name
+                    )
+                )
```

Comment:
The user wants to be informed there is not enough stock, but the error message breaks the transaction of partial reservation. Do you have a suggestion on how to solve it?

Reply:
My suggestion has been, for some time now, to only raise an exception when *none* of the lines of the picking cannot be (partially) assigned, instead of, as is currently the case, raise an exception when *one* of the lines of the picking cannot be (partially) assigned.

Reply:
@StefanRijnhart (Assuming you replied to me) I didn't mean a message in the exception, but in the chatter. I realize that the ticket is about there not being any feedback at all. I also realize that there is enough information as the message already mentions which moves are unreserved. I thought it would be a nice extra (I agree that we don't need it) to add to the message (that's already posted anyway) that a partial reservation happend.

Reply:
@dkubiak789 thanks for the refactoring!